pgmax/fg/graph.py

```diff
@@ -806,10 +910,23 @@ def BP(
         \tget_bp_state: Function to reconstruct the BPState from BPArrays.\n
         \tget_beliefs: Function to calculate beliefs from BPArrays.\n
     """
-    max_msg_size = int(np.max(bp_state.fg_state.wiring.edges_num_states))
-    num_val_configs = (
-        int(bp_state.fg_state.wiring.factor_configs_edge_states[-1, 0]) + 1
-    )
+    wiring = bp_state.fg_state.wiring
+    max_msg_size = int(np.max(wiring.edges_num_states))
```

Comment:
Calculate this on demand instead of having to define an additional `GraphWiring` class.